source/OctoVersion.Core/OutputFormatting/GitHubActions/GitHubActionsOutputFormatter.cs

```diff
@@ -36,21 +37,31 @@ public bool MatchesRuntimeEnvironment()
 
     static void WriteOutputVariables(OctoVersionInfo octoVersionInfo)
     {
-        // ::set-output name='octoversion_ddd'::'fff']
-
         var properties = octoVersionInfo.GetType()
             .GetProperties(BindingFlags.Public | BindingFlags.DeclaredOnly | BindingFlags.Instance);
-        foreach (var property in properties)
-        {
-            var configurationVariableKey = $"octoversion_{property.Name.ToLowerInvariant()}";
 
-            var value = property.GetValue(octoVersionInfo)?.ToString() ?? string.Empty;
+        // https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter
+        // The outgoing parameters must be written to a temporary file (identified by the $GITHUB_OUTPUT environment
+        // variable, which changes for every step in a workflow) which is then parsed. That file must also be UTF-8 or it will fail.
+        var gitHubOutputFilePath = System.Environment.GetEnvironmentVariable(GitHubActionsOutputFileEnvironmentVariableName);
 
-            var configurationVariableMessage = $"::set-output name={configurationVariableKey}::{value}";
-            System.Console.WriteLine(configurationVariableMessage);
+        if (gitHubOutputFilePath != null)
+        {
+            GitHubActionsLogSink.Log(LogEventLevel.Information, $"Writing version variables to {GitHubActionsOutputFileEnvironmentVariableName} file ({gitHubOutputFilePath}) for '{nameof(GitHubActionsOutputFormatter)}'.");
+            using var streamWriter = File.AppendText(gitHubOutputFilePath);
+            foreach (var property in properties)
+            {
+                var configurationVariableKey = $"octoversion_{property.Name.ToLowerInvariant()}";
+                var value = property.GetValue(octoVersionInfo)?.ToString() ?? string.Empty;
+                streamWriter.WriteLine($"{configurationVariableKey}={value}");
+            }
+        }
+        else
+        {
+            GitHubActionsLogSink.Log(LogEventLevel.Warning, $"Unable to write output parameters to ${GitHubActionsOutputFileEnvironmentVariableName} because the environment variable is not set.");

```

Comment:
```suggestion
            GitHubActionsLogSink.Log(LogEventLevel.Warning, $"Unable to write output parameters because the environment variable '{GitHubActionsOutputFileEnvironmentVariableName}' is not set.");

```